plugins/kubernetes/app/models/kubernetes/service.rb

```diff
@@ -0,0 +1,95 @@
+module Kubernetes
+  # Convenience wrapper around the 'Service' object we can get from the K8S API
+  class Service
+    attr_reader :deploy_group, :role, :name, :ip_address, :port_info
+
+    def initialize(deploy_group: nil, role: nil)
+      @deploy_group = deploy_group
+      @role = role
+      @port_info = []
+      @name = role.try(:service_name)
```

Comment:
Possibly.  But then the `Service` object would have no connection to the `Project`, which doesn't seem right to me.

I actually created a version of this code where `Service` was an `ActiveRecord::Base` subclass, and existed as a table in the database.  I'm still going back and forth on whether that's a good idea, so that you could more easily query all the Services running.  But since I wasn't sure about that, I started with the simplest implementation, which was this.